### DISCUSSION TITLE
improve testing experience

### DISCUSSION THREAD

**Discussion by JustFly1984** [emotion-js/emotion#2487] [Ideas]
I had an issue with testing emotion components with jest and @react-testing library, the case was missing `<ThemeProvider />` in the react component tree, but an  TypeError in tests was pointing to mangled classname access on undefined `styled.div()`. I've spend several hours debugging the case.

It was crashing for any emotion component, even if theme is not accessed.

Is it possible to make sensible check for ThemeProvider existence instead of crashing with unreadable error stack?

**Comment by Andarist** [comment]
Please always try to share a repro case in a runnable form - either by providing a git repository to clone or a codesandbox. OSS maintainers usually can't afford the time to set up the repro, even if exact steps are given.

**Reply by JustFly1984** [reply]
I can't share the repo, cos it is private. I've posted here so others could find a solution. I did an upgrade from 9 to 11, and tests was failing. Now I've fixed it, sorry, have no time for repro.

**Reply by JustFly1984** [reply]
The solution I propose, is to crash styled component with error, if ThemeProvider is not in the tree, as redux does.

**Reply by JustFly1984** [reply]
Try to remove <ThemeProvider from an app, and you will see what I'm talking about
